src/gc.c

```diff
@@ -55,41 +58,34 @@ typedef struct GarbageCollectorCtx {
 
   uint64_t spec_unique_id;
 
+  bool noLockMode;
+
 } GarbageCollectorCtx;
```

Comment:
We need to merge this and the `gc` object into one, or figure out how/why they are used differently.

Reply:
no, the gc object is an abstract object contains the ctx and callbacks to the real gc. This way we can implement number of gc techniques that implements the same callbacks (interface).

Reply:
So let's divide based on the callbacks and other stateful data. Or maybe call it GcForkImpl or ForkGC, or whatever.

A better approach would be to use an embedded base/overlay method, with something like

```c
typedef struct GC {
  void Start(GC *);
  void Stop(GC *);
  void OnDelete(GC *);
  ...
} GC;

// later on
typedef struct MyGCIMpl {
  GC base;
  void *myStuff
  ....
};

// and finally

static void MyGCImpl_Start(GC *ptrbase) {
  MyGCImpl *ptr = (MyGcIMPL *)ptrbase;
  ...
}
```

This is perfectly valid and involves less of a matryoshka doll-like setup of various dangling objects.

Reply:
I do not see much of a difference between what you suggested and my impl, we will still have to do something like gc->Start(gc). I think my impl is better because it does not obligate you to put unneeded variables inside your struct when implementing a new gc algorithm.